scrambles/src/puzzle/ThreeByThreeCubeFewestMovesPuzzle.java

```diff
@@ -17,7 +17,57 @@ public ThreeByThreeCubeFewestMovesPuzzle() {
 
     @Override
     public PuzzleStateAndGenerator generateRandomMoves(Random r) {
-        // TODO - explain tom2 here
+        // For fewest moves, we want to minimize the probability that the
+        // scramble has useful "stuff" in it. The problem with conventional
+        // Kociemba 2 phase solutions is that there's a pretty obvious
+        // orientation step, which competitors might intentionally (or even
+        // accidentally!) use in their solution. To lower the probability of this happening,
+        // we intentionally generate dumbed down solutions.
+
+        // We eventually decided to go with "Tom2", which is described by Tom
+        // Rokicki (https://groups.google.com/d/msg/wca-admin/vVnuhk92hqg/P5oaJJQjDQAJ):
+
+        // START TOM MESSAGE
+        // If we're going this direction, and the exact length isn't critical, why not combine a bunch of the
+        // ideas we've seen so far into something this simple:
+        //
+        // 1. Fix a set of prefix/suffix moves, say, U F R / U F R.
+        // 2. For a given position p, find *any* solution where that solution prefixed and suffixed with
+        //    the appropriate moves is canonical.
+        //
+        // That's it.  So we generate a random position, then find any two-phase solution g such
+        // that U F R g U F R is a canonical sequence, and that's our FMC scramble.
+        //
+        // The prefix/suffix will be easily recognizable and memorable and ideally finger-trick
+        // friendly (if it matters).
+        //
+        // Someone wanting to practice speed FMC (hehe) can make up their own scrambles just
+        // by manually doing the prefix/suffix thing on any existing scrambler.
+        //
+        // It does not perturb the uniformity of the random position selection.
+        //
+        // It's simple enough that it is less likely to suffer from subtle defects due to the
+        // perturbation of the two-phase search (unlikely those these may be).
```

Comment:
Also, it doesn't really make sense to say whether a specific scramble was perturbed. The distribution is a *little bit* perturbed for *all scrambles*.